zhaquirks/tuya/ts0601_sensor.py

```diff
@@ -1,487 +1,67 @@
-    }
-
-    data_point_handlers = {
-        3: "_dp_2_attr_update",
-        5: "_dp_2_attr_update",
-        15: "_dp_2_attr_update",
-    }
+(
+    TuyaQuirkBuilder("_TZE200_yjjdcqsq", "TS0601")
+    .also_applies_to("_TZE200_9yapgbuv", "TS0601")
+    .also_applies_to("_TZE204_yjjdcqsq", "TS0601")
+    .also_applies_to("_TZE200_utkemkbs", "TS0601")
+    .also_applies_to("_TZE204_utkemkbs", "TS0601")
+    .also_applies_to("_TZE204_yjjdcqsq", "TS0601")
+    .also_applies_to("_TZE204_ksz749x8", "TS0601")
+    .tuya_temperature(dp_id=1)
+    .tuya_humidity(dp_id=2)
+    .tuya_dp(
+        dp_id=4,
+        ep_attribute=TuyaPowerConfigurationCluster2AAA.ep_attribute,
+        attribute_name="battery_percentage_remaining",
+        converter=lambda x: {0: 50, 1: 100, 2: 200}[x],
+    )
+    .adds(TuyaPowerConfigurationCluster2AAA)
+    .add_to_registry()
+)
 
 
-class TuyaSoilSensorVar02(CustomDevice):
-    """Tuya temp and humidity sensor (variation 05)."""
+(
+    TuyaQuirkBuilder("_TZE284_aao3yzhs", "TS0601")
+    .also_applies_to("_TZE284_sgabhwa6", "TS0601")
+    .tuya_temperature(dp_id=5, scale=10)
+    .tuya_battery(dp_id=15)
+    .tuya_soil_moisture(dp_id=3)
+    .add_to_registry()
+)
 
-    signature = {
-        # "profile_id": 260,
-        # "device_type": "0x0051",
-        # "in_clusters": ["0x0000","0x0004","0x0005","0xed00","0xef00"],
-        # "out_clusters": ["0x000a","0x0019"]
-        MODELS_INFO: [
-            ("_TZE284_aao3yzhs", "TS0601"),
-            ("_TZE284_sgabhwa6", "TS0601"),
-        ],
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    0xED00,
-                    TemperatureHumidityManufCluster.cluster_id,
-                ],
-                OUTPUT_CLUSTERS: [Ota.cluster_id, Time.cluster_id],
-            }
-        },
-    }
 
-    replacement = {
-        SKIP_CONFIGURATION: True,
-        ENDPOINTS: {
-            1: {
-                DEVICE_TYPE: zha.DeviceType.TEMPERATURE_SENSOR,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    SoilManufClusterVar02,
-                    TuyaTemperatureMeasurement,
-                    TuyaSoilMoisture,
-                    TuyaPowerConfigurationCluster2AAA,
-                ],
-                OUTPUT_CLUSTERS: [Ota.cluster_id, Time.cluster_id],
-            }
-        },
-    }
+(
+    TuyaQuirkBuilder("_TZE200_myd45weu", "TS0601")
+    .also_applies_to("_TZE200_ga1maeof", "TS0601")
+    .also_applies_to("_TZE200_9cqcpkgb", "TS0601")
+    .also_applies_to("_TZE204_myd45weu", "TS0601")
+    .tuya_temperature(dp_id=5)
+    .tuya_battery(dp_id=15)
+    .tuya_soil_moisture(dp_id=3)
```

Comment:
Actually, is the scale correct on these?

Reply:
Temp and soil both default to a scale of 100, which matches the original quirk. Do we want to explicitly set the scale to avoid confusion?

Reply:
Ah, right. I got a bit confused looking at everything.
> Do we want to explicitly set the scale to avoid confusion?

Hmm, let's try not to for now (most seem to use `100`). If it causes more issues in the future, we can do that though.